packages/agreement-process/src/services/agreementService.ts

```diff
@@ -1131,6 +1149,48 @@ export function agreementServiceBuilder(
         await repository.createEvent(event);
       }
     },
+    async verifyTenantCertifiedAttributes(
+      {
+        tenantId,
+        descriptorId,
+        eserviceId,
+        delegationId,
+      }: {
+        tenantId: TenantId;
+        descriptorId: DescriptorId;
+        eserviceId: EServiceId;
+        delegationId?: DelegationId;
+      },
+      { logger, authData }: WithLogger<AppContext>
+    ): Promise<agreementApi.hasCertifiedAttributes> {
+      logger.info(
+        `Veryfing tenant ${tenantId} has required certified attributes for descriptor ${descriptorId} of eservice ${eserviceId}`
+      );
+
+      if (delegationId) {
+        const delegation = await retrieveDelegationById(
+          readModelService,
+          delegationId
+        );
+
+        assertDelegationIsActive(delegation);
+        assertIsDelegate(delegation, authData.organizationId);
+      } else {
```

Comment:
Also here and in all the validators, errors, etc. let's always explicitly mention the fact that it's a consumer delegation:  consumerDelegationId, `const consumerDelegation`, `assertIsDelegate` could have the kind as a parameter or become `assertIsDelegateConsumer`, etc...